src/control/system/checker/finding.go

```diff
@@ -0,0 +1,146 @@
+//
+// (C) Copyright 2022 Intel Corporation.
+//
+// SPDX-License-Identifier: BSD-2-Clause-Patent
+//
+
+package checker
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	chkpb "github.com/daos-stack/daos/src/control/common/proto/chk"
+)
+
+type (
+	Finding struct {
+		chkpb.CheckReport
+	}
+
+	reportObject uint
+)
+
+const (
+	poolObj reportObject = iota
+	contObj
+	targObj
+)
+
+func (ro reportObject) String() string {
+	return map[reportObject]string{
+		poolObj: "pool",
+		contObj: "container",
+		targObj: "target",
+	}[ro]
+}
+
+func (f *Finding) IsValidAction(action chkpb.CheckInconsistAction) bool {
```

Comment:
Is the question more like "HasAction"? Wondering what "valid" really means here.